### PR TITLE
Mv counters update

### PR DESCRIPTION
**Pull request by matthewvon**
This branch updates the performance counters Init() function to properly resize the shared memory as variables get added to the code.  Previously, a user would have to manually "ipcrm" the shared memory segment (or reboot the computer) to have counters work after leveldb upgrade.

perf_count_test was created to validate the changes.  It is now part of the standard "make check" tests.

### CODE REVIEW

File: util/perf_count.cc [hunk 1]
```diff
@@ -205,23 +206,63 @@
...
+        // does the shared memory already exists (and of proper size if writing)
+        should_create=(0!=ret_val || ((shm_info.shm_segsz < sizeof(PerformanceCounters)) && !IsReadOnly));
+
+        // should old shared memory be deleted?
+        if (should_create && 0==ret_val)
+        {
+            ret_val=shmctl(id, IPC_RMID, &shm_info);
+            if (0!=ret_val)
+            {
+                syslog(LOG_ERR, "shmctl IPC_RMID failed [%d, %m]", errno);
```
Comment (jonmeredith) [inline util/perf_count.cc:239]: Never used %m before, should this be 

```
syslog(LOG_ERR, "shmctl IPC_RMID failed [%d, %m]", errno, errno);
```

Or is errno read from TLS?

Comment (matthewvon) [inline util/perf_count.cc:239]: (from http://linux.die.net/man/3/printf)

m

(Glibc extension.) Print output of strerror(errno). No argument is required.

Comment (jonmeredith) [inline util/perf_count.cc:239]: Happily works on Linux/OSX too as written.

**Comment by jonmeredith** [comment]
+1 merge.

**Pull request merged**